src/linux/system.rs

```diff
@@ -520,14 +506,27 @@ impl SystemExt for System {
     }
 
     fn get_kernel_version(&self) -> Option<String> {
-        match read_to_string("/proc/version") {
-            Ok(s) => s.trim().split(' ').nth(2).map(String::from),
-            Err(_) => None,
+        let mut raw = std::mem::MaybeUninit::<libc::utsname>::zeroed();
+
+        if unsafe { libc::uname(raw.as_mut_ptr()) } == 0 {
+            let buffer: Vec<u8> = {
+                let info = unsafe { raw.assume_init() };
+
+                info.release
+                    .iter()
```

Comment:
Also, I think it'd be better to instead get the first null byte (if any) and then use https://doc.rust-lang.org/nightly/std/str/fn.from_utf8.html

Reply:
Resolving in favor of the first thread on this point.